packages/nimble-components/src/combobox/testing/combobox.pageobject.ts

```diff
@@ -0,0 +1,128 @@
+    }
+
+    // The Selection API currently isn't supported properly or consistently in shadow DOM,
+    // so unfortunately our delete API has to be passed the post-deletion text.
+    public setInputText(text: string, asDelete = false): void {
+        this.comboboxElement.control.value = text;
+        const inputEvent = new InputEvent('input', {
+            data: text,
+            inputType: asDelete ? 'deleteContentForward' : 'insertText'
+        });
+        this.comboboxElement.control.dispatchEvent(inputEvent);
+    }
+
+    /**
+     * Sets the input text and commits the value by pressing Enter. Will emit a 'change' event.
+     */
+    public commitValue(text: string): void {
+        this.setInputText(text);
+        this.pressEnterKey();
+    }
+
+    /**
+     * Either opens or closes the dropdown depending on its current state
+     */
+    public clickCombobox(): void {
+        this.comboboxElement.click();
+    }
+
+    public async clickAndWaitForOpen(): Promise<void> {
+        this.clickCombobox();
+        await this.regionLoadedListener.promise;
+        await waitForUpdatesAsync();
+        await waitAnimationFrame(); // necessary because scrolling is queued with requestAnimationFrame
+    }
+
+    /**
+     * Either opens or closes the dropdown depending on its current state
+     */
+    public clickDropdownButton(): void {
+        this.comboboxElement.dropdownButton!.control.click();
+    }
+
+    public async clickDropdownButtonAndWaitForOpen(): Promise<void> {
+        this.clickDropdownButton();
+        await this.regionLoadedListener.promise;
+        await waitAnimationFrame(); // necessary because scrolling is queued with requestAnimationFrame
+    }
+
+    public isDropdownButtonChecked(): boolean {
+        return this.comboboxElement.dropdownButton!.checked;
+    }
+
+    public isDropdownButtonDisabled(): boolean {
+        return this.comboboxElement.dropdownButton!.disabled;
+    }
+
+    /**
+     * Either opens or closes the dropdown depending on its current state
+     */
+    public clickInput(): void {
+        this.comboboxElement.control.click();
+    }
+
+    public getInputTitle(): string | null {
+        return this.comboboxElement.control.getAttribute('title');
+    }
+
+    public getInputAriaLabel(): string | null {
+        return this.comboboxElement.control.getAttribute('aria-label');
+    }
+
+    public async mouseoverInput(): Promise<void> {
+        this.comboboxElement.control.dispatchEvent(new MouseEvent('mouseover'));
+        await waitForUpdatesAsync();
+    }
+
+    public async mouseoutInput(): Promise<void> {
+        this.comboboxElement.control.dispatchEvent(new MouseEvent('mouseout'));
+        await waitForUpdatesAsync();
+    }
+
+    public async prependOption(value: string, label: string): Promise<void> {
+        this.comboboxElement.insertAdjacentHTML(
+            'afterbegin',
+            `<nimble-list-option value="${value}">${label}</nimble-list-option>`
```

Comment:
make sure to use the listOptionTagName instead of hard-coding. That ensure the element is imported and registered.